### ISSUE TITLE
Documentation: OSC URL changes in v3

### ISSUE THREAD

**Issue by bxtdvd** [cpvalente/ontime#892]
Thanks again for all the great work on this project!

I spent some time tonight debugging OSC input seemingly not working on v3, but it turns out the OSC URLs have just changed. :)

For my particular use case (showing video countdowns from ProPresenter and/or Mitti, via Bitfocus Companion's Generic OSC module):
`/ontime/set-external-message-text` is now `/ontime/message/external/text`
and
`/ontime/set-external-message-visible` is now `/ontime/message/external/visible`

I'm sure it's on the todo list, but just in case, these documentation pages will need updating:
https://docs.getontime.no/features/external-data/
https://docs.getontime.no/api/osc-and-ws

Let me know if there's any way I could be of assistance here. Thanks!

**Comment by cpvalente** [comment]
Hi @bxtdvd, thank you for this.

I apologise for the trouble caused, in retrospect, the breaking changes should have been better communicated. 
As of v3, all the APIs have changed (for the better I hope)

The documentation is trailing slightly behind but we hope to have it finished soon as we work with the companion module.

We have left gitbook and the documentation now lives in [a repo which users can track](https://github.com/cpvalente/ontime-docs). The specific pull request for updating the API documentation [is here](https://github.com/cpvalente/ontime-docs/pull/8/files)

If you have any other thoughts on the new docs or something doesnt add up, feel free to reach out

Thank you again

**Comment by cpvalente** [comment]
Hi @bxtdvd, we have merged today the changes to the API documentation.
Would you mind taking a look?

Here is the OSC docs
https://docs.getontime.no/api/protocols/osc/

If something is unclear I am happy to revise

**Comment by bxtdvd** [comment]
Thanks!

I think it would also be helpful to document the `/ontime/message/external/` OSC endpoint, too.

Also, [this page](https://docs.getontime.no/features/external-data/) still has the old API endpoints on it, and has a hyperlink ("see the docs") that is now a 404.

I know the info on that page is largely the same as what's on the API page, but I would argue in favor of keeping it around so people can see what the external data feature looks like and what it could be used for. I'm not sure if the updated version of the Companion plugin will allow setting the external data directly (or if you'll still need to use OSC for that), but I'd be happy to write out a better/updated version of [these instructions that I previously wrote up for somebody on Reddit](https://www.reddit.com/r/VIDEOENGINEERING/comments/18925w9/new_ontime_version_kicks_even_more_ass_free/kcicmj8/) and add them, with screenshots, as a pull request to the external data doc page.

**Comment by cpvalente** [comment]
Thank you @bxtdvd ,

It seems that I had misconfigured tooling to check our links and was having some unbased confidence in the documentation links. This has been fixed now

I had not seen your description in reddit. Thank you very much for helping people through in understanding the spaghetti (and often outdated) documentation we write

I hope that the new documentation is improved, but also a good step into making it easier for users to help.
If you feel the documentation can be improved, a PR or at least an issue is very much welcome
Specifically for the external message we currently have https://github.com/cpvalente/ontime-docs/issues/12
